plasmapy/classes/plasma_base.py

```diff
@@ -0,0 +1,26 @@
+from collections import OrderedDict
+from abc import ABC
+
+# GenericPlasma subclass registry
+PLASMA_CLASSES = OrderedDict()
+
+class GenericPlasmaRegistrar(ABC):
```

Comment:
I am assuming that you are planning on using this class for API definition and not just to be the keeper of the registry, so I would suggest you rename it to `BasePlasma` or something similar. As this is the class you will want to do `isinstance` with to test if it implements your plasma API

Reply:
I don't know for sure if we are going to use this class for API definition or not.

Based on my understanding, `GenericPlasma` is supposed to deal with data that is common in all plasmas. If we are going to have `GenericPlasmaRegistrar` do this task as well then we could remove `GenericPlasma` class and have this class renamed to `BasePlasma` and deal with both common plasma data and do the registration thing.

Thoughts @SolarDrew, @namurphy?

Reply:
It's a good idea to have an Abstract base class that dosen't implement any (much) API and then a base class which implements the abstract class in a way which is generic for plasmapy. This means that if anyone wants to do something *very* different down the line they can have all the benifits of the ABC and the registration without having to worry about any of the implementation in the generic class.

For an example of this see [ndcube](https://github.com/sunpy/ndcube/blob/master/ndcube/ndcube.py#L26)

Reply:
I think I understand now, that ndcube example is very interesting. So, we do both the registration thing and *declare* some abstract methods on the `BasePlasma` but define them in `GenericPlasma`. I'm good with that.